adapters/mqtt-vertx/src/main/java/org/eclipse/hono/adapter/mqtt/impl/IMessageMapping.java

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package org.eclipse.hono.adapter.mqtt.impl;
+
+import org.eclipse.hono.adapter.mqtt.MqttContext;
+import org.eclipse.hono.util.ResourceIdentifier;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.mqtt.messages.MqttPublishMessage;
+
+/**
+ * A message mapping interface. This component requests mapping from another server is configured properly. The
+ * headers are overwritten with the result of the mapper (which includes the resourceId).
+ * E.g.: when the deviceId is in the payload of the message, the deviceId can be deducted in the custom mapper and
+ * the payload can be changed accordingly to the payload originally received by the gateway.
+ */
+public interface IMessageMapping {
```

Comment:
We do not use this naming pattern to indicate an interface. This is simply the *MessageMapping* interface and your implementation might be the *HttpBasedMessageMapping* or something indicating its particular strategy for mapping the messages.